js/tooltip.js

```diff
@@ -385,13 +385,54 @@
     return delta
   }
 
+  Tooltip.prototype.didModuleReceiveTypeStringOrFunction = function (module, received) {
+    if (!/string|function/.test(received)) {
+      throw new Error(module + ' received type ' + received + '. Was expecting type string or a function.')
+    }
+
+    return true
+  }
+
+  Tooltip.prototype.didFunctionReturnStringOrJQuery = function (module, returned, returnedType) {
+    if (returnedType !== 'string' && !(returned.jquery)) {
+      throw new Error(module + ' received type function which returned type ' + returnedType + '. ' +
+        'Was expecting type string as the return type.')
+    }
+
+    return true
+  }
+
+  Tooltip.prototype.checkValidType = function (module, contentType, o) {
+    var returnedContent
+    this.didModuleReceiveTypeStringOrFunction(module, contentType)
+
+    if (contentType == 'function') {
+      if (module === 'Popover') {
+        returnedContent = o.content.call(this.$element[0])
+      }
+      if (module === 'Tooltip') {
+        returnedContent = o.title.call(this.$element[0])
+      }
```

Comment:
And what about directly passing in a DOM node or a jQuery object instead of a function returning either of those?